File: Лаб_2/OP_Lab_2/OP_Lab_2/Patient.cpp
#include "Patient.h"

Patient::Patient() {
	this->name = "";
	this->surname = "";
	this->lastVisitDate;
	this->curVisitDate;
}

void Patient::displayData()
{
	cout << this->name << endl;
	cout << this->surname << endl;
	cout << this->lastVisitDate << endl;
	cout << this->curVisitDate << endl << endl;
}

Patient::Patient(string name, string surname, Date lastVisitDate, Date curVisitDate) {
	this->name = name;
	this->surname = surname;
	this->lastVisitDate = lastVisitDate;
	this->curVisitDate = curVisitDate;
}


bool Patient::isNewPatient()
{
	const int days_from_last_visit = 10;
	return (this->curVisitDate.convertToDays() - this->lastVisitDate.convertToDays() > days_from_last_visit);
}

bool Patient::timeIsUp()
{
	time_t t = time(0);
	tm* now = localtime(&t);
	Date currDate(now->tm_mday, now->tm_mon + 1, now->tm_year + 1900);
		
	return currDate.convertToDays() > this->curVisitDate.convertToDays();
}

//bool Patient::timeIsUp()
//{
//	/*time(NULL);
//	return ;*/
//}

string Patient::getName() { return (this->name); }
string Patient::getSurname() { return (this->surname); }
Date Patient::getLastVisitDate() { return (this->lastVisitDate); }
Date Patient::getCurVisitDate() { return (this->curVisitDate); }


void Patient::setName(string name) { this->name = name; }
void Patient::setSurname(string surname) { this->surname = surname; }
void Patient::setLastVisitDate(Date lastVisitDate) { this->lastVisitDate = lastVisitDate; }
void Patient::setCurVisitDate(Date curVisitDate) { this->curVisitDate = curVisitDate; }

ostream& operator<<(ostream& os, const Patient& patient)
{
	/*os.write((char*)&(patient.name), sizeof(patient.name));
	os.write((char*)&(patient.surname), sizeof(patient.surname));

	os.write((char*)&(patient.lastVisitDate), sizeof(patient.lastVisitDate));
	os.write((char*)&(patient.curVisitDate), sizeof(patient.curVisitDate));*/

	os << patient.name << endl << patient.surname << endl << patient.lastVisitDate << endl << patient.curVisitDate << endl << endl;
	return os;
}

istream& operator>>(istream& is, Patient& patient) {
	/*is.read((char*)&patient.name, sizeof(patient.name));
	is.read((char*)&patient.surname, sizeof(patient.surname));
	is.read((char*)& patient.lastVisitDate, sizeof(patient.lastVisitDate));
	is.read((char*)&patient.curVisitDate, sizeof(patient.curVisitDate));*/
	
	is >> patient.name >> patient.surname >> patient.lastVisitDate >> patient.curVisitDate;
	return is;
}

File: Лаб_2/OP_lab_2_py/Patient.py
import datetime


class Patient:
    def __init__(self, surname, last_visit_date, appointment_time ):
        self.__surname = surname
        self.__last_visit_date = datetime.datetime.strptime(last_visit_date, "%d.%m.%Y")
        self.__appointment_datetime = datetime.datetime.strptime(appointment_time, "%d.%m.%Y %H:%M")

    def __str__(self):
        return "Surname: {0}\nLast vist date: {1}\nAppointment date and time: {2}\n".format(self.__surname, self.__last_visit_date.strftime("%d.%m.%Y"), self.__appointment_datetime.strftime("%d.%m.%Y %H:%M"))

    def time_is_up(self):
        return datetime.datetime.now() > self.__appointment_datetime

    def is_recent_patient(self):
        delta_days = datetime.timedelta(10)
        return datetime.datetime.now() - delta_days <= self.__last_visit_date

    def get_surname(self):
        return self.__surname

    def get_last_visit_date(self):
        return self.__last_visit_date

    def get_appointment_datetime(self):
        return self.__appointment_datetime

    def set_surname(self, surname):
        self.__surname = surname

    def set_last_visit_date(self, last_visit_date):
        self.__last_visit_date = last_visit_date

    def set_appointment_datetime(self, appointment_datetime):
        self.__appointment_datetime = appointment_datetime


